src/main/java/com/networknt/schema/ItemsValidator.java

```diff
@@ -123,9 +123,6 @@ public Set<ValidationMessage> walk(JsonNode node, JsonNode rootNode, String at,
             JsonNode defaultNode = null;
             if (applyDefaultsStrategy.shouldApplyArrayDefaults() && schema != null) {
                 defaultNode = schema.getSchemaNode().get("default");
-                if (defaultNode != null && defaultNode.isNull()) {
-                    defaultNode = null;
-                }
```

Comment:
This part was an open question.  If you are removing these lines, then it means you are applying default even if it is null in the case of a json array.  So then for parallel it make sense to do this for the other file where the
```
            boolean applyDefault = propertyNode == null
                    || (applyDefaultsStrategy.shouldApplyPropertyDefaultsIfNull() && propertyNode.isNull() && !defaultNode.isNull());
```
should be changed to
```
            boolean applyDefault = propertyNode == null || propertyNode.isNull()  && applyDefaultsStrategy.shouldApplyPropertyDefaultsIfNull();
```

On the other hand if you want to skip applying default if the default is null node, then keep line lines the other fille as is but bring back lines 126-128. 

Will let you decide how to want to go, but keep things in parallel.

Reply:
@SiemelNaran I have aligned the logic at both the places, by setting null even if value provided is null, because of your listener logic that may come in future for some requirement.